### PR TITLE
@rdfjs/environment v1 and dependencies

### PR DESCRIPTION
**Pull request by tpluscode**
Excuse the large-ish PR. I wanted to update `@rdfjs/environment` to v1 and following the thread I had to also update

1. rename `@rdfjs/formats-common` to `@rdfjs/formats`
2. update `@rdfjs/serializer-jsonld-ext` to v4
3. add `@rdfjs/serializer-turtle`
4. update `rdf-ext` to v2.5

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: 
   - [`@rdfjs/serializer-jsonld-ext` v4](https://github.com/rdfjs-base/serializer-jsonld-ext/commit/90450008e3b96eef48b20250a5a1487659dbba10#diff-7cfafc50d542656187fb90da3fb17f41011b4cb4f9b0f49a0af65c167a432e3e)
   - [`@rdfjs/formats`](https://github.com/rdfjs-base/formats/commit/eb9833a9dd6b045547fe22736ac45be16f695320)
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the `package.json`.

### CODE REVIEW

**Comment by tpluscode** [comment]
I renamed `rdfjs__formats-common` to `rdfjs__formats` because that package changed name. Now the CI tries to run tests for the former. What do I do?

@jakebailey, maybe you can help, please

**Comment by jakebailey** [comment]
It seems like we're just not detecting via the diff that an entire package was effectively renamed; we git diff to make sure that we can catch partial deletions and so on. It's likely that there's just something wrong. We could precheck for the dir and ignore it if it's gone? @sandersn

That being said, can you just keep the old package? https://www.npmjs.com/package/@rdfjs/formats-common still exists and is not deprecated; stuff typically isn't dropped from DT until it's been typed or is just entirely broken. (We don't yet have an "exit" criteria that includes "old and unused"). Then, the PR would pass.

**Comment by tpluscode** [comment]
> That being said, can you just keep the old package?

Now that you mention, it probably makes sense to keep it. One might still want to refine types of the old package. I'll do that. Thanks

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/rdfjs__serializer-jsonld-ext/index.d.ts
1. types/rdfjs__formats/index.d.ts
1. types/rdfjs__formats/pretty.d.ts
1. types/rdf-ext/index.d.ts
1. types/rdf-ext/FetchFactory.d.ts

<details>
<summary>as well as these 4 other files...</summary>
<p>types/rdf-ext/FormatsFactory.d.ts, types/rdfjs__environment/rdfjs__environment-tests.ts, types/rdfjs__serializer-turtle/index.d.ts, types/rdfjs__serializer-turtle/rdfjs__serializer-turtle-tests.ts</p>
</details>


Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 27468aa9990471b895b057fa015b18b014a643d7
</p>

**Comment by sandersn** [comment]
@jakebailey I'm pretty sure the code that reads `git diff` output got confused by codes R100, R071, R068, R083, R089 (see below).

I confirmed that the code only handles A, M and D. I'm not even sure what R*nnn* means, but it sounds like I need to learn.

```
R100	types/rdfjs__formats-common/.npmignore	types/rdfjs__formats/.npmignore
R100	types/rdfjs__environment/FormatsFactory.d.ts	types/rdfjs__formats/Factory.d.ts
R071	types/rdfjs__formats-common/index.d.ts	types/rdfjs__formats/index.d.ts
R100	types/rdfjs__environment/lib/Formats.d.ts	types/rdfjs__formats/lib/Formats.d.ts
R068	types/rdfjs__formats-common/package.json	types/rdfjs__formats/package.json
A	types/rdfjs__formats/pretty.d.ts
R083	types/rdfjs__formats-common/rdfjs__formats-common-tests.ts	types/rdfjs__formats/rdfjs__formats-tests.ts
A	types/rdfjs__formats/tsconfig.json
M	types/rdfjs__serializer-jsonld-ext/index.d.ts
M	types/rdfjs__serializer-jsonld-ext/package.json
M	types/rdfjs__serializer-jsonld-ext/rdfjs__serializer-jsonld-ext-tests.ts
A	types/rdfjs__serializer-turtle/.npmignore
A	types/rdfjs__serializer-turtle/index.d.ts
A	types/rdfjs__serializer-turtle/package.json
A	types/rdfjs__serializer-turtle/rdfjs__serializer-turtle-tests.ts
R089	types/rdfjs__formats-common/tsconfig.json	types/rdfjs__serializer-turtle/tsconfig.json
```

**Comment by sandersn** [comment]
Filed https://github.com/microsoft/DefinitelyTyped-tools/issues/901 to track the failure to understand git moves. I'll try to get to it this week or next.

File: types/rdf-ext/index.d.ts [hunk 1]
```diff
@@ -1,31 +1,28 @@
...
 import { TermMapFactory } from "@rdfjs/term-map/Factory.js";
 import { TermSetFactory } from "@rdfjs/term-set/Factory.js";
 import { TraverserFactory } from "@rdfjs/traverser/Factory.js";
-import ClownfaceFactory from "./ClownfaceFactory.js";
 import { DataFactoryExt as DataFactory } from "./DataFactory.js";
 import { DatasetFactory } from "./DatasetFactory.js";
 
 export { default as Environment } from "@rdfjs/environment/Environment.js";
-export { default as FormatsFactory } from "@rdfjs/environment/FormatsFactory.js";
 export { default as FetchFactory } from "@rdfjs/fetch-lite/Factory.js";
```
Comment (sheetalkamat) [inline types/rdf-ext/index.d.ts:14]: Dont you need to use newly added d.ts file instead ?

**Review by sheetalkamat — CHANGES_REQUESTED** [review]

File: types/rdf-ext/index.d.ts (hunk 1 above)
Comment (tpluscode) [reply types/rdf-ext/index.d.ts:14]: You mean `FormatsFactory`? I export it on line 23

**Review by sheetalkamat — APPROVED** [review]

**Comment by tpluscode** [comment]
Ready to merge

**Pull request merged**